### PR TITLE
Subjects workbooks

### PR DESCRIPTION
**Pull request by anastasiiatulentseva**
- add subject selection for teacher
- add smart index for teacher and students
- add special sign up fields for students

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -1,16 +1,29 @@
 class RegistrationsController < Devise::RegistrationsController
 
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :get_specialties, only: [:new, :create, :edit]
+
+  def create
+    unless params[:user][:is_student]
+      params[:user].delete(:student_profile_attributes)
+    end
```
Comment (stulentsev) [inline app/controllers/registrations_controller.rb:9]: what is this for? :)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -21,10 +21,27 @@ def index
     end
   end
 
-  def update
+  def set_subjects
```
Comment (stulentsev) [inline app/controllers/users_controller.rb]: rename to a more fitting `choose_subjects`, for example.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -21,10 +21,27 @@ def index
...
+    @user = User.find(params[:id])
+    @subjects = Subject.for_teacher_or_unassigned(@user.teacher_profile)
+    @grouped_subjects = @subjects.group_by{|s| s.specialty.name }
+  end
+
+  def save_subjects
+    @user = User.find(params[:id])
+    @subjects = Subject.for_teacher_or_unassigned(@user.teacher_profile)
+    @grouped_subjects = @subjects.group_by{|s| s.specialty.name }
```
Comment (stulentsev) [inline app/controllers/users_controller.rb:33]: You can shorten to `@subjects.group_by(&:specialty_name)`

File: app/controllers/workbooks_controller.rb [hunk 4]
```diff
@@ -1,20 +1,37 @@
...
     @workbook = Workbook.find(params[:id])
   end
 
   def index
-    @workbooks = Workbook.all
+    if current_user.teacher?
+      @subjects = @user.teacher_profile.subjects
+      @workbooks = @user.teacher_profile.workbooks
+    else
+      @subjects = Subject.where(specialty_id: @user.student_profile.specialty_id, course_year: @user.student_profile.course_year)
```
Comment (stulentsev) [inline app/controllers/workbooks_controller.rb]: extract scope(s)

File: app/controllers/workbooks_controller.rb [hunk 5]
```diff
@@ -1,20 +1,37 @@
...
+    end
+
+    subject_name = params[:subject]
+    if subject_name
+      pill_subject = Subject.find(params[:subject])
+      @panel_workbooks = @workbooks.select { |w| w.subject_id == pill_subject.id }
+      @panel_heading = pill_subject.name
+    else
+      @panel_workbooks = @workbooks
+      @panel_heading = 'All'
```
Comment (stulentsev) [inline app/controllers/workbooks_controller.rb:25]: i18n?

File: db/seeds.rb [hunk 6]
```diff
@@ -10,3 +11,16 @@
...
+7.times do |n|
+  Specialty.create!(name: Faker::Educator.campus)
+end
+
+#subjects
+specialties = Specialty.all
+8.times do |n|
+  name = Faker::Educator.course
+  course_year = Faker::Number.between(1, 6)
+  specialties.each {|specialty| specialty.subjects.create!(name: name, course_year: course_year )}
```
Comment (stulentsev) [inline db/seeds.rb]: I think multiline here would read better.

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (anastasiiatulentseva) [reply app/controllers/users_controller.rb:33]: doesn't work (

**Pull request merged**